### PR TITLE
Update deployment scripts and README's

### PR DESCRIPTION
**Pull request by davidz627**
Improved deployment script to build in container and reduce image size from 216MB to 6.9MB. Updated READMEs

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -25,18 +25,16 @@ gce-pd-driver:
 	go build -o bin/gce-pd-csi-driver-test ./test/e2e/
 
 build-container: gce-pd-driver
-	cp bin/gce-pd-csi-driver deploy/docker
-	docker build -t $(STAGINGIMAGE):$(STAGINGVERSION) deploy/docker
+	docker build -t $(STAGINGIMAGE):$(STAGINGVERSION) .
```
Comment (msau42) [inline Makefile]: do you also want to remove references to your dev project and make that an env variable?

File: README.md [hunk 2]
```diff
@@ -10,5 +10,40 @@ Specification compliant driver used by Container Orchestrators to manage the
...
+
+## Dependency Management
+Use [dep](https://github.com/golang/dep)
+```
+$ dep ensure
+```
+
+To modify dependencies or versions change `./Gopkg.toml`
+
+[deployREADME]: github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/deploy/kubernetes/README.md
```
Comment (msau42) [inline README.md]: You can do relative references here

File: Makefile (hunk 1 above)
Comment (davidz627) [reply Makefile]: done

File: README.md (hunk 2 above)
Comment (davidz627) [reply README.md]: done

**Comment by msau42** [comment]
/lgtm
In the future, you may want to check out kompose for generating dev vs prod environments

**Comment by davidz627** [comment]
Fixed #4

**Pull request merged**